packages/node_modules/@webex/internal-plugin-encryption/src/encryption.js

```diff
@@ -120,11 +121,12 @@ const Encryption = WebexPlugin.extend({
     const inputBody = {
       endpoints: [scr.loc]
     };
+    const endpointUrl = url.parse(scr.loc);
+    const serviceUrl = `${endpointUrl.protocol}//${endpointUrl.hostname}/v1/download/endpoints`;
```

Comment:
I don't think we need to declare the endpoint.protocol? Everything should be over https and I wouldn't want us trying to retrieve anything over http. Feels like there might be a better way to do this via the service links but maybe not. Will defer to @InteractiveTimmy. This seems like a simple solution if it works.

Reply:
Done!

Reply:
@andrewholsted  -  I removed the `protocol`, but leads to code failure. Seems like protocol is mandatory for `uri`. So, I am reverting it.
@InteractiveTimmy - any inputs.. ?

Reply:
@Akashdeep-Sarin, if as @andrewholsted states that we don't want to retrieve anything over HTTP, you can hardcode the protocol to HTTPS, so it would only look for resources over HTTPS

Reply:
If the hostname is not always from the home cluster [i.e. it is possible to get an EU hostname when you're on US clusters in this use case], then this may be the best way to perform these actions. the URL dependency can be used to make this process a bit easier and clearer.

```js
endpointUrl.host = ...;
endpointUrl.protocol  = ...;
endpoint.path = ...;
endpointUrl = url.format(endpointUrl);
```

See Node's documentation on URL [here](https://nodejs.org/api/url.html#url_url_strings_and_url_objects)